src/main/java/de/uslu/ShopService.java

```diff
@@ -12,8 +13,8 @@ public ShopService(OrderRepo myOrders, ProductRepo myProducts) {
         this.myProducts = myProducts;
     }
 
-    public Product getProduct(String id) {
-        return myProducts.get(id);
+    public Optional<Product> getProduct(String id) {
+        return Optional.ofNullable(myProducts.get(id));
```

Comment:
You should place the Optional in your Repo Class, so you can check int the Service if it´s empty and adapt to it. Right now your just taking the optional without unwrapping it.